lib/editor-linter.coffee

```diff
@@ -1,102 +1,42 @@
-{CompositeDisposable, Emitter, Range} = require 'atom'
-Helpers = require './helpers'
+{TextEditor, Emitter, CompositeDisposable} = require('atom')
 
 class EditorLinter
-  constructor: (@linter, @editor) ->
-    @status = true
-    @messages = new Map
-    @inProgress = false
-    @inProgressFly = false
-
-    if @editor is atom.workspace.getActiveTextEditor()
-      @linter.views.updateLineMessages(true)
-
+  constructor: (@editor) ->
+    throw new Error("Given editor isn't really an editor") unless @editor instanceof TextEditor
+    @status = true # Overall linting status
     @emitter = new Emitter
     @subscriptions = new CompositeDisposable
-
-    @subscriptions.add(
-      @editor.onDidSave => @lint(false)
-    )
-    @subscriptions.add(
-      @editor.onDidChangeCursorPosition ({oldBufferPosition, newBufferPosition}) =>
+    @subscriptions.add @editor.onDidDestroy =>
+      @emitter.emit 'did-destroy'
+
+    @subscriptions.add @editor.onDidSave => @emitter.emit('should-lint', false)
+    # The onDidStopChanging callbacks are invoked immediately, and we want to avoid it
+    setImmediate =>
+      @subscriptions.add @editor.onDidStopChanging => setImmediate => @emitter.emit('should-lint', true)
```

Comment:
```CoffeeScript
emitShouldLint = =>
    # The onDidStopChanging callbacks are invoked immediately, and we want to avoid it
    setImmediate => @emitter.emit('should-lint', true)
    # Alternatively this will also work and doesn't require an intermediate function just for binding.
    # setImmediate @emitter.emit.bind(this, 'should-lint', true)


@subscriptions.add @editor.onDidStopChanging(emitShouldLint)
```

This separates registering the callback from the definition of it and makes it easier to see the two pieces.